### DISCUSSION TITLE
How to show linting and type errors in web server

### DISCUSSION THREAD

**Discussion by axhamre** [expo/expo#14370] [Archived: Questions] [answered]
I'm running a web project with expo, and wonder how I could get the same kind of typing/linting logging that you get with CRA?

**Comment by EvanBacon** [comment, marked as answer]
Currently we _nerf_ Webpack to make it work more like the Metro bundler (React Native default) which doesn't support custom features like this, but we plan to improve the entire bundler ecosystem in Expo and make all platforms support linting/type errors and warnings. 

You can still manually add this feature to the web platform by ejecting the `webpack.config.js` (by running `expo customize:web`), then add the `eslint-webpack-plugin` to the config (similar to how [CRA does](https://github.com/facebook/create-react-app/blob/b45ae3c9caf10174d53ced1cad01a272d164f8de/packages/react-scripts/config/webpack.config.js#L26) and [here](https://github.com/facebook/create-react-app/blob/b45ae3c9caf10174d53ced1cad01a272d164f8de/packages/react-scripts/config/webpack.config.js#L741-L765)).

Be sure to set the cache location to `path.join(env.projectRoot, '.expo', 'lint')` (lint can be anything), this will ensure that clearing works as expected. 

For typing, it could be a little bit more complicated since the Metro babel preset strips types automatically.

**Reply by axhamre** [reply]
Amazing, thanks!
